tests/migrations/test_optimizer.py

```diff
@@ -270,6 +270,35 @@ def test_optimize_through_create(self):
             app_label="testapp",
         )
 
+        # This could be optimized a bit more but at least it generates a valid
+        # set of operations.
+        self.assertOptimizesTo(
+            [
+                migrations.CreateModel('Book', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Person', [('name', models.CharField(max_length=255))]),
+                migrations.AddField('book', 'author', models.ForeignKey('test_app.Person', models.CASCADE)),
+                migrations.CreateModel('Review', [('book', models.ForeignKey('test_app.Book', models.CASCADE))]),
+                migrations.CreateModel('Reviewer', [('name', models.CharField(max_length=255))]),
+                migrations.AddField('review', 'reviewer', models.ForeignKey('test_app.Reviewer', models.CASCADE)),
+                migrations.RemoveField('book', 'author'),
+                migrations.DeleteModel('Person'),
+            ],
+            [
+                migrations.CreateModel('Person', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Book', [
+                    ('name', models.CharField(max_length=255)),
+                    ('author', models.ForeignKey('test_app.Person', models.CASCADE)),
+                ]),
+                migrations.CreateModel('Reviewer', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Review', [
+                    ('book', models.ForeignKey('test_app.Book', models.CASCADE)),
+                    ('reviewer', models.ForeignKey('test_app.Reviewer', models.CASCADE)),
+                ]),
+                migrations.RemoveField('book', 'author'),
```

Comment:
I actually don't think we need the field context for `RemoveField`? Since the executor doesn't seem to need the context (this operation works, after all).

Even though `author` is a column that is a foreign key to another table, the `RemoveField` operation itself doesn't use this information. Just the information that the field is on `book`.